packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -196,29 +190,297 @@ class CupertinoAlertDialog extends StatelessWidget {
+                children: <Widget>[
+                  new CupertinoLayoutId(
+                    id: _AlertDialogSection.content,
+                    child: _buildContent(),
+                  ),
+                  new CupertinoLayoutId(
+                    id: _AlertDialogSection.actions,
+                    child: _buildActions(),
+                  ),
+                ],
+              ),
+            ),
+          ),
         ),
       ),
     );
   }
 }
 
+enum _AlertDialogSection {
+  content,
+  actions,
+}
+
+class CupertinoDialogRenderWidget extends MultiChildRenderObjectWidget {
+  CupertinoDialogRenderWidget({
+    Key key,
+    @required List<Widget> children,
+  }) : super(key: key, children: children);
+
+  @override
+  RenderObject createRenderObject(BuildContext context) {
+    return new CupertinoDialogRenderBox();
+  }
+}
+
+class CupertinoDialogRenderBox extends RenderBox
+    with ContainerRenderObjectMixin<RenderBox, CupertinoDialogRenderBoxParentData>,
+        RenderBoxContainerDefaultsMixin<RenderBox, CupertinoDialogRenderBoxParentData> {
+  CupertinoDialogRenderBox({
+    List<RenderBox> children,
+  }) {
+    addAll(children);
+  }
+
+  @override
+  void setupParentData(RenderBox child) {
+    if (child.parentData is! CupertinoDialogRenderBoxParentData)
+      child.parentData = new CupertinoDialogRenderBoxParentData();
+  }
+
+  @override
+  double computeMinIntrinsicWidth(double height) {
+    RenderBox _content;
+    RenderBox _actions;
+    getChildrenAsList().forEach((RenderBox child) {
+      MultiChildLayoutParentData parentData = child.parentData;
+      if (parentData.id == _AlertDialogSection.content) {
+        _content = child;
+      } else if (parentData.id == _AlertDialogSection.actions) {
+        _actions = child;
+      }
+    });
+    assert(_content != null);
+    assert(_actions != null);
+
+    final double contentWidth = _content.getMaxIntrinsicWidth(height);
+    final double actionsWidth = _actions.getMaxIntrinsicWidth(height);
+    final double width = max(contentWidth, actionsWidth);
+
+    if (width.isFinite)
+      return width;
+    return 0.0;
+  }
+
+  @override
+  double computeMaxIntrinsicWidth(double height) {
+    RenderBox _content;
+    RenderBox _actions;
+    getChildrenAsList().forEach((RenderBox child) {
+      CupertinoDialogRenderBoxParentData parentData = child.parentData;
+      if (parentData.id == _AlertDialogSection.content) {
+        _content = child;
+      } else if (parentData.id == _AlertDialogSection.actions) {
+        _actions = child;
+      }
+    });
+    assert(_content != null);
+    assert(_actions != null);
+
+
+    final double contentWidth = _content.getMaxIntrinsicWidth(height);
+    final double actionsWidth = _actions.getMaxIntrinsicWidth(height);
+    final double width = min(contentWidth, actionsWidth);
+
+    if (width.isFinite)
+      return width;
+    return 0.0;
+  }
+
+  @override
+  double computeMinIntrinsicHeight(double width) {
```

Comment:
Let's extract a common utility for this that takes a function argument (min or max)

Reply:
I'm not sure what you mean, can you elaborate?